public/app/features/alerting/unified/components/receivers/DuplicateTemplateView.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { FC } from 'react';
+
+import { Alert } from '@grafana/ui';
+import { AlertManagerCortexConfig } from 'app/plugins/datasource/alertmanager/types';
+
+import { updateAndSanitizeDefine } from '../../utils/templates';
+
+import { TemplateForm } from './TemplateForm';
+
+interface Props {
+  templateName: string;
+  config: AlertManagerCortexConfig;
+  alertManagerSourceName: string;
+}
+
+export function generateCopiedTemplateName(config: AlertManagerCortexConfig, originalTemplateName: string): string {
```

Comment:
I wonder if we could extract some parts of this function that overlap with the `generateCopiedRuleTitle` code.
The new name generation code is not trivial, and it would be nice not to duplicate it

Reply:
💯